strings/base_activation.h

```diff
@@ -47,6 +43,21 @@ namespace winrt::impl
             return 0;
         }
 
+        // If the class is not registered, and regfree support is enabled, give that a chance to make the activation succeed.
+        // We should not attempt to fallback to regfree for error codes besides class not registered.  If the activation is out
+        // of process then we could have RPC error codes.  It would be bad if a class that should only ever be used out of proc
+        // is erroneously activated inproc.
+        if ((hr == error_class_not_registered) && reg_free_factory_getter)
```

Comment:
Matching on specific error codes is notoriously hard to do reliably. How sure are you that this won't fail with other error codes like `CLASS_E_CLASSNOTAVAILABLE` or `REGDB_E_CLASSNOTREG`?

Reply:
`REGDB_E_CLASSNOTREG` was already covered by `winrt::impl::error_class_not_registered`.  Good call about `CLASS_E_CLASSNOTAVAILABLE` though.  Pushed an iteration where that is covered too (via `winrt::impl::error_class_not_available`).

Reply:
Actually, reverted this change.  I asked a COM expert what HRESULTs would be appropriate to check here and that is `REGDB_E_CLASSNOTREG` only.

`CLASS_E_CLASSNOTAVAILABLE` is returned when the registration exists and points to a DLL but then that DLL says "I don't implement that class".  In that case we should not fall back to regfree behavior because the registration does exist but is not usable.